src/dotnet/Parser.cs

```diff
@@ -3,14 +3,23 @@
 
 using Microsoft.DotNet.Cli.CommandLine;
 using Microsoft.DotNet.Tools.Help;
+using Microsoft.DotNet.Tools.New;
 using static System.Environment;
 using static Microsoft.DotNet.Cli.CommandLine.LocalizableStrings;
 using LocalizableStrings = Microsoft.DotNet.Tools.Run.LocalizableStrings;
+using NewCommandParser = Microsoft.TemplateEngine.Cli.CommandParsing.CommandParserSupport;
 
 namespace Microsoft.DotNet.Cli
 {
     public static class Parser
     {
+        // This is used for descriptions of commands and options that are only defined for `dotnet complete` (i.e. command line completion).
+        // For example, a NuGet assembly handles parsing the `nuget` command and options.
+        // To get completion for such a command, we have to define a parser that is used for the completion.
+        // Command and option help text cannot be empty, otherwise the parser will hide them from the completion list.
+        // The value of `-` has no special meaning; it simply prevents these commands and options from being hidden.
```

Comment:
seems like '-' is not the original design of complete. This is not ideal. Can we do something on dotnet complete layer not to hide the command? and why parser will hide no description?

Reply:
There is no way to control this that is external to the parser.  It's simply how it defines "hidden" options and commands currently.